hummingbot/market/binance/binance_time.py

```diff
@@ -66,24 +71,25 @@ def stop(self):
             self._time_offset_ms.clear()
             self._started = False
 
-    async def set_server_time_offset(self):
+    async def update_server_time_offset_loop(self):
         while True:
-            try:
-                async with aiohttp.ClientSession() as session:
-                    async with session.get(self.BINANCE_TIME_API) as resp:
-                        time_now_ms = time.time() * 1e3
-                        resp_data = await resp.json()
-                        binance_server_time = resp_data["serverTime"]
-                        time_after_ms = time.time() * 1e3
-                expected_server_time = int((time_after_ms + time_now_ms)//2)
-                time_offset =  binance_server_time - expected_server_time
-                self.set_time_offset_ms(time_offset)
-
-            except asyncio.CancelledError:
-                raise
-            except Exception:
-                self.logger().network(f"Error getting Binance server time.", exc_info=True,
-                                      app_warning_msg=f"Could not refresh Binance server time. "
-                                                      f"Check network connection.")
+            await self.update_server_time_offset()
+            await asyncio.sleep(self._server_time_offset_check_interval)
 
-            await asyncio.sleep(self.SERVER_TIME_OFFSET_CHECK_INTERVAL)
+    async def update_server_time_offset(self):
```

Comment:
This function will likely get called too many times within a short window when we get TimeStamp errors burst. You can add last_updated_timestamp and check if it is still within, says, 5 seconds, just return.